### ISSUE TITLE
etcd sends 307 redirects to PUT requests

### ISSUE THREAD

**Issue by stevenschlansker** [etcd-io/etcd#673]
```
[etcd] Apr  1 14:45:40.320 DEBUG     | [recv] PUT http://Anesthetize.local:4002 /v2/keys/ot/srvc/default/55bbb813-2bae-4a87-a817-644ab0b5b1c3 [127.0.0.1:53715]
[etcd] Apr  1 14:45:40.320 DEBUG     | Redirect to http://Anesthetize.local:4003/v2/keys/ot/srvc/default/55bbb813-2bae-4a87-a817-644ab0b5b1c3
```

Compliant HTTP libraries are not happy about this.
From http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

> If the 307 status code is received in response to a request other than GET or HEAD, the user agent MUST NOT automatically redirect the request unless it can be confirmed by the user, since this might change the conditions under which the request was issued.

**Comment by brentryan** [comment]
Can I assume this is fixed in etcd 2.0 as well?

**Comment by xiang90** [comment]
@brentryan Yes.
